File: 4_greedy/이준우/단속 카메라.py
def solution(routes):
    routes.sort()

    # 초깃값 routes[1]을 선택
    left, right = routes[0]
    answer = 1
    for l, r in routes:
        # 핵심 풀이 : 하나의 단속 카메라가 많은 차량을 커버하기 위한 범위를 좁혀나감
        # l만 체크하는 이유 : l순으로 정렬되어 있기 때문에 모든 수 커버 가능
        if left <= l <= right:
            left = max(left, l)
            right = min(right, r)
        else:
            answer += 1
            left = l
            right = r

    return answer


"""
컨셉 : 하나의 단속 카메라가 많은 차량을 커버하기 위해 범위를 좁혀나감
"""

File: 4_greedy/김민주/단속카메라.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes) {
    int n = routes.size();
    int answer = 0;
    
    vector <pair <int, int>> rt;
    for (int i=0; i<n; i++){
        // rt : {나감, 들어옴}
        rt.push_back({routes[i][1], routes[i][0]});
    }
    // 2번째 값(나간시점)으로 정렬
    sort(rt.begin(), rt.end());
    
    int idx =0;
    int cctv = rt[idx].second -1;
    while (idx < n){
        if (cctv < rt[idx].second){
            // 새 cctv 
            cctv = rt[idx].first;
            answer += 1;
        }
        idx +=1;
    }
    
    
    return answer;
}